src/Microsoft.DotNet.Interactive/Events/DisplayedValue.cs

```diff
@@ -35,6 +36,11 @@ public void Update(object updatedValue)
                 _mimeType,
                 updatedValue.ToDisplayString(_mimeType));
 
+            if (KernelInvocationContext.Current?.Command is SubmitCode)
+            {
+                _context = KernelInvocationContext.Current;
```

Comment:
This is kind of weird. 

What happens in Jupyter?

Reply:
The behaviour is consistent and the data is updated on the frontend. We tested that too